src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -66,10 +66,20 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
               std::is_same<typename Node::Tag, NodeWithLeftChildAndRopeTag>{},
           "Unrecognized node tag");
 
-      Kokkos::parallel_for("ArborX::TreeTraversal::spatial",
-                           Kokkos::RangePolicy<ExecutionSpace>(
-                               space, 0, Access::size(predicates)),
-                           *this);
+#if defined(KOKKOS_ENABLE_CUDA) && KOKKOS_VERSION >= 30300
```

Comment:
Please add a comment to capture what you discuss in the description of the PR, that is why you chose this even though you could in principle have expressed the preference for all execution spaces.